airbyte-cdk/python/airbyte_cdk/sources/declarative/datetime/datetime_parser.py

```diff
@@ -25,8 +25,11 @@ def parse(self, date: Union[str, int], format: str, timezone):
         # See https://stackoverflow.com/a/4974930
         if format == "%s":
             return datetime.datetime.fromtimestamp(int(date), tz=timezone)
-        else:
-            return datetime.datetime.strptime(str(date), format).replace(tzinfo=timezone)
+
+        parsed_datetime = datetime.datetime.strptime(str(date), format)
+        if self._is_naive(parsed_datetime):
+            return parsed_datetime.replace(tzinfo=timezone)
+        return parsed_datetime
```

Comment:
I don't really like the interface of this method since timezone is not necessarily used. For example, if I call:
`DatetimeParser. parse("2023-04-27T06:29:50-01:00", "%Y-%m-%dT%H:%M:%S%z", datetime.timezone.utc)`, it feels like the returned datetime should have the timezone UTC but it won't. Therefore, I'll remove the parameter completely

Reply:
With the most recent change where we only set the timezone if the datetime is naive, I could call `DatetimeParser.parse("2023-04-27T06:29:50-01:00", "%Y-%m-%dT%H:%M:%S%z", datetime.timezone.utc)` and since the datetime provide is not naive, it would keep the timezone from the datetime. Hence, the parameter `datetime.timezone.utc` is confusing. Therefore, I removed the timezone from the parameters of the method. Anyway, I don't think it added any value since the value of this parameter was always UTC

Reply:
ah okay so you were talking about the method signature change for the whole method. I think I was just confused because I thought this was in reference to the exact line of this comment. I agree getting rid of the parameter makes sense. thanks for clarifying!

Reply:
Yes, sorry for the confusion. I made changes after writing this comment which might have generated even more confusion